docs/walk-through/dags.md

```diff
@@ -0,0 +1,234 @@
+# DAGs
+
+DAGs (Directed Acyclic Graphs) are similar to `Steps`, and also use a context manager in the same way. DAGs are formed
+of `Tasks`, and offer more flexibility in Workflow construction, with the key difference being that you can specify the
```

Comment:
I found the ordering a little confusing: it wasn't immediately obvious that you were still comparing DAGs to Steps by the time you got to "the key difference". I also thought "can specify" might lead people to think it's optional. How about this?

```suggestion
DAGs (Directed Acyclic Graphs) are collections of `Tasks`. They are similar to `Steps`, and use a context manager in the
same way, but offer more flexibility in Workflow construction, with the key difference being that you must specify the
```

Reply:
Thanks, rejigged the paragraph a bit